device_doctor/lib/src/device.dart

```diff
@@ -0,0 +1,92 @@
+// Copyright 2020 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'config.dart';
+import 'ios_device.dart';
+
+/// The root of the API for controlling devices.
+DeviceDiscovery get devices => DeviceDiscovery();
+
+/// Operating system on the devices that this host is configured to test.
+enum DeviceOperatingSystem { android, ios }
+
+/// Discovers available devices and chooses one to work with.
+abstract class DeviceDiscovery {
+  factory DeviceDiscovery() {
+    switch (config.deviceOperatingSystem) {
+      case DeviceOperatingSystem.ios:
+        return IosDeviceDiscovery();
+      default:
+        throw StateError('Unsupported device operating system: {config.deviceOperatingSystem}');
+    }
+  }
+
+  /// Lists all available devices' IDs.
+  Future<List<Device>> discoverDevices({int retriesDelayMs = 10000});
```

Comment:
Swarming bots are usually associated with one DUT. Is there a use case for Flutter to support multiple devices from one bot?

Reply:
We currently only support one device on one bot, but may support more to one in the future when needed.

Reply:
A host can run multiple swarming bots, but a swarming bot should only be connected to one DUT. Since this is a swarming bot utility, shouldn't this be phrased for a single device?

The counterexample I listed was moreso if there is a test that requires more than one phone. I've seen those set ups for swarming where they created specialized pools to support those use cases.